### ISSUE TITLE
How to install the visual components? Directory "\Libraries\bpl" does not exist.

### ISSUE THREAD

**Issue by ViaCriativa** [MagicFoundation/Alcinoe#283]
Hello, how to install the visual components?
Directory '\Libraries\bpl" does not exist.
Thanks!!

**Comment by Zeus64** [comment]
hello, you need to run CompileAll.bat (https://github.com/MagicFoundation/Alcinoe/blob/master/CompileAll.bat)

**Comment by DiamatBR** [comment]
mine gave the following error
![image](https://user-images.githubusercontent.com/77128586/235226436-cd2457db-b53d-4fef-bf9a-789c9634cc13.png)

**Comment by Zeus64** [comment]
ouch something look very bad, could you add here all your alcinoe-master directory ?

**Comment by DiamatBR** [comment]
I was trying to use version 1.00 and it didn't work.
When I download version 1.04... I add it to the list of libraries... I open the source and try to compile it to be able to use VideoPlayer, I get this error:
![image](https://user-images.githubusercontent.com/77128586/235260538-3a7431dc-c2fe-4f5d-94f5-75bb9c7147fa.png)

**Comment by DiamatBR** [comment]
> ouch something look very bad, could you add here all your alcinoe-master directory ?

Where do you want me to add it? In the project? Or on Github?

**Comment by Zeus64** [comment]
Ok, regarding the last error it's because you need Alexandria 11.3, it's seam you have alexandria 11.2

**Comment by DiamatBR** [comment]
Estou usando o Delphi 10.4 CE. Do you have any tips to resolve this?

**Comment by Zeus64** [comment]
you can use [chatGPT ](https://chat.openai.com/) if you are not very well in english :) you need the last delphi Alexandria version 11.3 CE

**Comment by DiamatBR** [comment]
Hi! What I did was the following:

1- Install Delphi 11 CE
2- Download Alcinoe
3- I unpacked
4- Build all projects in the Tools folder
5- In Delphi Tools > Option > Language > Delphi > Library > Platform Android 32 bits > Library path
Add: "C:\Users\DiamatBR\Desktop\Desenvolvimento\Delphi\Componentes\Alcinoe-master\Source"
6- Run CompileAll.bat
-type "1" <ENTER>
-typed "Y" <ENTER>
-waited to finish
7- Run CompileJar.bat
8- Open AlcinoeAlexandria in Delphi:
-right-clicked on the project and ordered install AlcinoeAlexandria.bpl
7- Created a new application:
-put a TALVideoPlayerSurface:
-select Android 32-bit
-had it run in debug mode, and...

**ERROR**
"Exception class EJNI with message 'Java typer JExoPlayer_Builder could nor be found'. Process RP_InDoorVideoAndro.apk (5692)"
![image](https://user-images.githubusercontent.com/77128586/235377331-21c424bf-21d8-4d8d-acfe-b39e5218c5cf.png)

**Comment by Zeus64** [comment]
that normal, it's because you need to add the exoplayer JAR libraries. 
have a look at this demo: https://github.com/MagicFoundation/Alcinoe/tree/master/Demos/ALFmxControls/_Source

and especially this batch : https://github.com/MagicFoundation/Alcinoe/blob/master/Demos/ALFmxControls/_Source/android/MergeLibraries.bat

**Comment by DiamatBR** [comment]
I believe that even after taking a look at ALFmxControls I could not evolve in that. I don't think it works in delphi 11 CE
![image](https://user-images.githubusercontent.com/77128586/235517980-d4b88ea4-c2f5-42ed-a277-58079f579474.png)

**Comment by Zeus64** [comment]
yes i think delphi 11 CE is base on 11.0 not on 11.3 :(

**Comment by DiamatBR** [comment]
So I won't be able to use Alcinoe :'(
I have Delphi 10.4 CE

**Comment by campbell-gary** [comment]
Thank you DiamatBR for posting the steps for installing Alcinoe. At least now I can see the Alcinoe components on the Delphi Palette. I have Delphi 11.3.1 and I compiled an app for Android 64 with a TALVideoPlayerSurface component, but the app hangs in the splash screen.

**Comment by campbell-gary** [comment]
When I place a TALMemo component on a form and compile for Android64, the app hangs on the splash screen. When I compile with the debugger I get the following error messages: 

- Project MemoTest.apk raised exception class Segmentation fault (11)
- Source file not found: ZygoteInit.java

**Comment by Zeus64** [comment]
@campbell-gary can you try to run this demo: https://github.com/MagicFoundation/Alcinoe/tree/master/Demos/ALFmxControls/_Source ?

**Comment by campbell-gary** [comment]
Fantastic. The demo ALFmxControls compiled and ran on my phone without any problem using the Android64 platform. What changes do you recommend I make?

**Comment by Zeus64** [comment]
you must do something similar than this on your app : https://github.com/MagicFoundation/Alcinoe/blob/master/Demos/ALFmxControls/_Source/android/MergeLibraries.bat

**Comment by campbell-gary** [comment]
I opened the MergeLibraries.bat on my laptop but it was missing the assignment statements "set Libraries=", so I copied them and added them to my MergeLibraries.bat. I compiled a test app using the Android64, platform, but the app still hangs on the splash screen.

Messages received:
- Project Memotest.apk raised exception class Segmentation fault (11)
- A messagebox displays: Source file not found - Path to source file ZygoteInit.java
 
Your FMX ALFmxControls compile and run fine.
